server/src/modules/auth/auth.service.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  BadRequestException,
+  ForbiddenException,
+  Injectable,
+} from '@nestjs/common';
+import { AuthDto } from './dto/auth.dto';
+import { CreateUserDto } from '../users/dto';
+import { UsersService } from '../users/users.service';
+import { RefreshTokensService } from '../refresh-token/refresh-tokens.service';
+import * as bcrypt from 'bcrypt';
+import { authServiceErrorMessages } from './auth.constants';
+import { AuthTokens, JwtPayload } from './types';
+import { JwtService } from '@nestjs/jwt';
+import { AppConfigService } from 'src/config/app-config.service';
+import { UserEntity } from '../users/user.entity';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private usersService: UsersService,
+    private refreshTokensService: RefreshTokensService,
+    private jwtService: JwtService,
+    private appConfigService: AppConfigService,
+  ) {}
+
+  async signUp(createUserDto: CreateUserDto): Promise<any> {
+    const userExists = await this.usersService
+      .findOne({ email: createUserDto.email })
+      .then(() => true)
+      .catch(() => false);
```

Comment:
We expect object when calling this.usersService() and returning null when there is probles or object is not find is common practice 
```suggestion
      .catch(() => null);
```